### PR TITLE
 Add support for specifying a spec file for deployments

### PR DESCRIPTION
**Pull request by birgirst**

### CODE REVIEW

File: fiaas_skipper/__init__.py [hunk 1]
```diff
@@ -34,21 +38,39 @@ def init_k8s_client(config):
...
+            release_channel_factory = FakeReleaseChannelFactory(json.loads(cfg.release_channel_metadata))
+        else:
+            release_channel_factory = ReleaseChannelFactory(cfg.baseurl)
+        if os.path.isfile(cfg.spec_file):
+            try:
+                spec_config = _load_spec_config(cfg.spec_file)
+                log.debug("Loaded spec config from file {!r}".format(cfg.spec_file))
+            except yaml.YAMLError:
+                log.exception("Unable to load spec config file {!r} using defaults".format(cfg.spec_file))
+                spec_config = None
```
Comment (mortenlj) [inline fiaas_skipper/__init__.py]: If you set `spec_config=None` before the if-test, you can drop the else-block and this line, making the flow easier to follow.

Comment (birgirst) [reply fiaas_skipper/__init__.py]: Good point. I'll add that.

File: fiaas_skipper/__init__.py [hunk 2]
```diff
@@ -34,21 +38,39 @@ def init_k8s_client(config):
...
+
+
 def main():
     cfg = Configuration()
     init_logging(cfg)
     init_k8s_client(cfg)
     log = logging.getLogger(__name__)
     try:
-        log.info("fiaas-skipper starting with configuration {!r}".format(cfg))
+        log.info("essgettittt fiaas-skipper starting with configuration {!r}".format(cfg))
```
[the remark this answers is not in the document]
Comment (birgirst) [reply fiaas_skipper/__init__.py]: Stray debug token that shouldn't have been part of the commit. I'll fix.

**Comment by birgirst** [comment]
I'll rebase

**Pull request merged**